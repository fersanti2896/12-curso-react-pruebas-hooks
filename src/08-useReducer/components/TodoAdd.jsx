import { useForm } from '../../hooks';

export const TodoAdd = ({ onNewTodo }) => {
    const { onInputChange, description, onResetForm } = useForm({
        description: ''
    });

    const onFormSubmit = ( event ) => {
        event.preventDefault();

        if( description.length <= 1 ) return;

        const newTodo = {
            id: new Date().getTime(),
            done: false,
            description
        }

        onNewTodo( newTodo );
        onResetForm();
    }

    return (
        <>
            <form onSubmit={ (event) => onFormSubmit(event) }>
                <input type='text'
                       className='form-control'
                       placeholder='¿Qué hay que hacer?'
                       name='description'
                       onChange={ onInputChange }
                       value={ description } />

                <button className='btn btn-outline-success mt-2'
                        type='submit' >
                    Agregar
                </button>
            </form>
        </>
    )
}
