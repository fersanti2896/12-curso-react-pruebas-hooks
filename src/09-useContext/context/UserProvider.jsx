import { useState } from 'react';
import { UserContext } from './UseContext';

/* const user = {
    id: 123,
    name: 'Fernando NicolÃ¡s',
    email: 'fersanti2896@gmail.com'
}*/

export const UserProvider = ({ children }) => { 
    const [user, setUser] = useState();

    return (
        // <UserContext.Provider value={{ hola: 'mundo', user }} >
        <UserContext.Provider value={{ user, setUser }} >
            { children }
        </UserContext.Provider>
    )
}
