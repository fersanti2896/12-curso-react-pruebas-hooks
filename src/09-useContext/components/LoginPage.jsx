import { useContext } from 'react';
import { UserContext } from '../context/UseContext';

export const LoginPage = () => {
    const { user, setUser } = useContext( UserContext );

    return (
        <>
            <h1>LoginPage</h1>
            <hr/>

            <pre aria-label='pre'>
                { JSON.stringify( user, null, 3 ) }
            </pre>

            <button className='btn btn-outline-success'
                    onClick={ () => setUser({ id: 123, name: 'Fernando', email: 'prueba@gmail.com' }) }>
                Establecer usuario
            </button>
        </>
    )
}
