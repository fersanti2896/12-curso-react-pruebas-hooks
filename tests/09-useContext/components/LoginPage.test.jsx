import { fireEvent, render, screen } from '@testing-library/react';
import { LoginPage } from '../../../src/09-useContext/components/LoginPage';
import { UserContext } from '../../../src/09-useContext/context/UseContext';

describe('Pruebas en <LoginPage />', () => {
    const user = {
        id: 1,
        name: 'Fer Santi',
        email: 'fersanti2896@gmail.com'
    } 

    test('Debe hacer match con el snapshot.', () => { 
        const { container } = render( <UserContext.Provider value={{ user }}> 
                                        <LoginPage />
                                      </UserContext.Provider>);

        expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar el componente sin el usuario.', () => { 
        render( <UserContext.Provider value={ {user: null} }> 
                    <LoginPage />
                </UserContext.Provider>);

        const preTag = screen.getByLabelText('pre').innerHTML;
        expect( preTag ).toBe('null');
    });

    test('Debe de llamar el setUser cuando se hace click en el boton.', () => { 
        const setUserMock = jest.fn();

        render( <UserContext.Provider value={{ user, setUser: setUserMock }}> 
                    <LoginPage />
                </UserContext.Provider>);
        
        const userButton = screen.getByRole('button', { name: 'Establecer usuario' });
        fireEvent.click( userButton );

        expect( setUserMock ).toHaveBeenCalled();
        expect( setUserMock ).toHaveBeenCalledWith( { id: 123, name: 'Fernando', email: 'prueba@gmail.com' } );
    });
});