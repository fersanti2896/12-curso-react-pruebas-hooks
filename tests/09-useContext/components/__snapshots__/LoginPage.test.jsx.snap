// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginPage /> Debe hacer match con el snapshot. 1`] = `
<div>
  <h1>
    LoginPage
  </h1>
  <hr />
  <pre
    aria-label="pre"
  >
    {
   "id": 1,
   "name": "Fer Santi",
   "email": "fersanti2896@gmail.com"
}
  </pre>
  <button
    class="btn btn-outline-success"
  >
    Establecer usuario
  </button>
</div>
`;
